plans/phase-1/admission-controller-and-defaults.md

```diff
@@ -0,0 +1,453 @@
+  * **The User-Facing Confusion:** Because these Level 2 defaults are not persisted back to `etcd`, a user running `kubectl get` on a CR sees a spec that does **not** match the functional state the operator is enforcing. This is confirmed by a 6-year-old `TODO` in their code explicitly mentioning the lack of mutating webhook support in Operator SDK as a blocker.
+  * **Our Motivation:** This "invisible" default pattern is unacceptable for `multigres-operator`. We *must* make sure our manifests are fully declarative and the user sees the complete CR with `kubectl get`. A **mutating webhook** is the correct solution, as it applies all defaults and persists them to `etcd` *before* the reconciler ever sees the object.
+
+### Motivation 2: The Need for Powerful, Multi-Layered Validation
+
+Our API design requires validation logic that is too complex for a single tool.
+
+  * **Simple Cross-Field Logic:** We need to enforce rules like mutual exclusion (e.g., "you can't set both `deploymentTemplate` and `managedSpec`") or logic (e.g., "etcd replicas must be an odd number"). This is best handled *inside* the CRD.
+  * **Context-Aware Logic:** We need to enforce rules that require request-time context, such as `request.userInfo` (to make child CRs read-only) or `oldObject.status` (to prevent an update from breaking quorum). This requires a `ValidatingAdmissionPolicy`.
+  * **Stateful Logic:** We need to enforce rules that query *other objects*. This includes protecting a `DeploymentTemplate` from deletion if a `MultigresCluster` is using it, or checking if a `DeploymentTemplate` exists on `MultigresCluster` creation. This logic can *only* be performed by a webhook.
+
+-----
+
+## Goals
+
+  * Implement a **mutating webhook** to persist complex defaults (e.g., image tags, resource blocks) into the `MultigresCluster` spec, solving the "invisible defaults" problem.
+  * Implement a **stateful validating webhook** for rules that require cluster-awareness (querying other objects), such as:
+      * Preventing the deletion of a `DeploymentTemplate` that is in use by any `MultigresCluster`.
+      * Validating that a referenced `DeploymentTemplate` exists upon `MultigresCluster` creation or update.
+  * Utilize **CRD-Embedded CEL (`XValidation`)** for all advanced *stateless, object-only* validation (e.g., mutual exclusion, odd numbers, required sub-fields).
+  * Utilize **`ValidatingAdmissionPolicy` (CEL)** for *stateless, context-aware* validation that needs access to `oldObject.status` or `request.userInfo`.
+  * Provide clear, actionable error messages at admission time.
+  * Support multiple certificate management strategies (cert-manager and self-contained).
+  * Keep webhook logic in the `data-handler` module.
```

Comment:
what is the distinction of the data-handler and why here instead of resource-handler?